src/actions/commands/actions.ts

```diff
@@ -2203,17 +2203,15 @@ class ActionDeleteCharWithDeleteKey extends BaseCommand {
   runsOnceForEachCountPrefix = true;
   canBeRepeatedWithDot = true;
 
-  public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    // N<del> is a no-op in Vim
-    if (vimState.recordedState.count !== 0) {
-      return vimState;
-    }
-
-    const state = await new operator.DeleteOperator(this.multicursorIndex).run(vimState, position, position);
-
-    state.currentMode = ModeName.Normal;
-
-    return state;
+  public async execCount(position: Position, vimState: VimState): Promise<VimState> {
+      if (vimState.recordedState.count !== 0) {
+        vimState.recordedState.count = Math.floor(vimState.recordedState.count / 10);
```

Comment:
http://vimdoc.sourceforge.net/htmldoc/change.html#<Del>

What the <Del> key does if you have a count in front of it is to remove the last number from the count. So if you typed 100<Del>x, then you would only delete 10 characters.

So I simply divide the count by 10 (which removes the last number).